deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvStages.java

```diff
@@ -103,6 +103,10 @@ public void update(@Context SecurityContext sc,
         if (origBean.getStage_type() != EnvType.DEFAULT && origBean.getStage_type() != environBean.getStage_type()) {
             throw new TeletaanInternalException(Response.Status.BAD_REQUEST, "Modification of stage type is not allowed!");
         }
+        if(environBean.getSOX()) {
+            // TODO: future - should a more granular resource be authorized on?
+            authorizer.authorize(sc, new Resource(environBean.getEnv_name(), Resource.Type.ENV), Role.SYSTEMADMIN);
```

Comment:
Not sure about the authorization, seems like we are authorizing on too broad an environment. Might be worth a closer look in the future.

Reply:
Are we going to reject PUT API call for all SOX env in general? This will break clients. 
We should reject only if  non SYSADMIN user is trying to modify SOX property.

Reply:
Good point, I will add a condition to check if is_sox is different than on origBean.